### ISSUE TITLE
Accessibility issue in Fluent AutoComplete

### ISSUE THREAD

**Issue by PraleshGitHub** [microsoft/fluentui-blazor#1292]
# 🐛 Bug Report

Accessibility compliance issues in fluent autocomplete.

## 💻 Repro or Code Sample
1. go to below url https://www.fluentui-blazor.net/Autocomplete
2. click on search icon of autocomplete fluent control
3. open NVDA
4. on keyboard press CAPS+n and go to speech viewer
5. open the autocomplete and go through the populated list
## 🤔 Expected Behavior

Screen reader NVDA should pronounce the correct role
should pronounce whether the autocomplete is expand or collapsed 
should go to the list box appear on search on tab click
should pronounce the list items

## 😯 Current Behavior

Roles not getting pronounce in NVDA, 
does not pronounce whether the autocomplete is expand or collapsed 
does not go to the list box appear on search on tab click
does not pronounce the list items

## 💁 Possible Solution

Search field should have aria label and roles defined. 
List options should be accessible by keyboard and should screen reader should narrate the list items 

## 🔦 Context
NA

## 🌍 Your Environment

* OS & Device: Windows
* Browser : Microsoft Edge, Google Chrome, Apple Safari, Mozilla FireFox]
* .NET: NET 6  and FAST Version : 3.5.0(microsoft.fast.components.fluentui)

**Comment by vnbaaij** [comment]
I'm not familiar with NVDA. Is it this? https://www.nvaccess.org/download/

**Comment by dvoituron** [comment]
Yes, the common tools used for Accessibility are:
- https://accessibilityinsights.io/downloads/
- https://www.nvaccess.org/download
- https://support.microsoft.com/en-us/windows/complete-guide-to-narrator-e4397a0d-ef4f-b386-d8ae-c172f109bdb1
- https://youtu.be/qDm7GiKra28
- https://youtu.be/_1yRVwhEv5I

But, sorry, at the moment, some of our components (including FluentAutocomplete) are not fully compatible with Accessiblity rules.

**Comment by PraleshGitHub** [comment]
> I'm not familiar with NVDA. Is it this? https://www.nvaccess.org/download/

yes this is NVDA tool, the given link can be used for NVDA download. However this issue is in general we see, you can use any narrator tool to check.

**Comment by PraleshGitHub** [comment]
> Yes, the common tools used for Accessibility are:
> 
> * https://accessibilityinsights.io/downloads/
> * https://www.nvaccess.org/download
> * https://support.microsoft.com/en-us/windows/complete-guide-to-narrator-e4397a0d-ef4f-b386-d8ae-c172f109bdb1
> * https://youtu.be/qDm7GiKra28
> * https://youtu.be/_1yRVwhEv5I
> 
> But, sorry, at the moment, some of our components (including FluentAutocomplete) are not fully compatible with Accessiblity rules.

is this in your pipeline?, when can we expect FluentAutocomplete control compliant with accessibility as other controls.

**Comment by vnbaaij** [comment]
Yes, we want to do this but we are a small team with a lot to do. Might take some time. We do take PR's so if you want to contribute that would be very much appreciated.

**Comment by PraleshGitHub** [comment]
@vnbaaij downloaded code locally. Checked the autocomplete component, it has been observed that the role combobox is getting assigned to fluent-text-field however its not getting passed to the input field generated when autocomplete is getting used.
if the role to input field is passed then narrator will pronounce correctly.

Since It looks like fluent-text-field is coming from some other library we may not be able to make changes.

Need your help to take it further for making this improvement.

We are also investigating the list part of autocomplete. We will come back with more details.

Thanks for looking into this.

**Comment by dvoituron** [comment]
Could you tell me which attributes we should add and I'll update the code.
If possible, with a final HTML example of the added attributes.

## Example:
- `<input role="combobox" ... />`
    see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role

**Comment by PraleshGitHub** [comment]
Hi @dvoituron, 
please add below attributes to input element
<input role="combobox" aria-controls>
aria-controls here pointing to FluentAnchoredRegion whereas it should point to div where role="listbox" is given. You need to give the div an id="listbox-1" and use this to set input aria-controls value when popup of autocomplete is open.

We have seen that the focus is not shifting in the listbox when popup has opened and on click of down arrow, narrator just keep pronouncing blank.

Tried to give tabindex=0 to FluentOption field in listbox, then narrator pronounce the option on hitting tab from keyboard however that may not be good practice.

Could you please check that part also.

Thanks for all your support.

**Comment by PraleshGitHub** [comment]
> Hi @dvoituron, please add below attributes to input element aria-controls here pointing to FluentAnchoredRegion whereas it should point to div where role="listbox" is given. You need to give the div an id="listbox-1" and use this to set input aria-controls value when popup of autocomplete is open.
> 
> We have seen that the focus is not shifting in the listbox when popup has opened and on click of down arrow, narrator just keep pronouncing blank.
> 
> Tried to give tabindex=0 to FluentOption field in listbox, then narrator pronounce the option on hitting tab from keyboard however that may not be good practice.
> 
> Could you please check that part also.
> 
> Thanks for all your support.

@dvoituron please add both role="combobox" and aria-controls attribute is also needed on input element, in my previous comment I have added but it's not visbible somehow.

**Comment by dvoituron** [comment]
Will be fixed in next release: #1374

**Comment by dvoituron** [comment]
Just after our quick call with Pralesh (thanks to you). We discovered another problem, related to the "position" of the pressed keys ⁠
Pressing the "separate" arrows is fine, but not using the NumPad arrows. 
The code used in C# is different. we aregoing to change this today and it will be included in the next update.

![image](https://github.com/microsoft/fluentui-blazor/assets/8350694/dd7a6e78-3527-4f3c-8d93-121040af0445)

**Comment by dvoituron** [comment]
Fix included in the PR #1383... Will be included in the next release.
